agent/agentssh/portinspection_supported.go

```diff
@@ -11,41 +11,42 @@ import (
 	"golang.org/x/xerrors"
 )
 
-func getListeningPortProcessCmdline(port uint32) (string, error) {
+/**
+ * getListeningPortProcessCmdlines looks up the full command line for all
+ * processes listening on the specified port regardless of their address.  It
+ * returns all the commands it was able to find and any errors encountered while
+ * doing the search.
+ */
+func getListeningPortProcessCmdlines(port uint32) ([]string, error) {
 	acceptFn := func(s *netstat.SockTabEntry) bool {
 		return s.LocalAddr != nil && uint32(s.LocalAddr.Port) == port
```

Comment:
Don't you want to check the state is listening?  Otherwise you'll get existing connections to the port which will return the agent process